File: library/db/src/main/java/org/keynote/godtools/android/db/FollowupMapper.java
package org.keynote.godtools.android.db;

import android.content.ContentValues;
import android.database.Cursor;

import org.cru.godtools.model.Followup;

import androidx.annotation.NonNull;

import static org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_CREATE_TIME;
import static org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_DESTINATION;
import static org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_EMAIL;
import static org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_LANGUAGE;
import static org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_NAME;

final class FollowupMapper extends BaseMapper<Followup> {
    @Override
    protected void mapField(@NonNull final ContentValues values, @NonNull final String field,
                            @NonNull final Followup followup) {
        switch (field) {
            case COLUMN_NAME:
                values.put(field, followup.getName());
                break;
            case COLUMN_EMAIL:
                values.put(field, followup.getEmail());
                break;
            case COLUMN_LANGUAGE:
                values.put(field, serialize(followup.getLanguageCode()));
                break;
            case COLUMN_DESTINATION:
                values.put(field, followup.getDestination());
                break;
            case COLUMN_CREATE_TIME:
                values.put(field, serialize(followup.getCreateTime()));
                break;
            default:
                super.mapField(values, field, followup);
                break;
        }
    }

    @NonNull
    @Override
    protected Followup newObject(@NonNull final Cursor c) {
        return new Followup();
    }

    @NonNull
    @Override
    public Followup toObject(@NonNull final Cursor c) {
        final Followup followup = super.toObject(c);

        followup.setName(getString(c, COLUMN_NAME));
        followup.setEmail(getString(c, COLUMN_EMAIL));
        followup.setLanguageCode(getLocale(c, COLUMN_LANGUAGE));
        followup.setDestination(getLong(c, COLUMN_DESTINATION));
        followup.setCreateTime(getDate(c, COLUMN_CREATE_TIME));

        return followup;
    }
}

File: library/db/src/main/java/org/keynote/godtools/android/db/FollowupMapper.kt
package org.keynote.godtools.android.db

import android.content.ContentValues
import android.database.Cursor
import org.ccci.gto.android.common.util.database.getLocale
import org.ccci.gto.android.common.util.database.getLong
import org.ccci.gto.android.common.util.database.getString
import org.cru.godtools.model.Followup
import org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_CREATE_TIME
import org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_DESTINATION
import org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_EMAIL
import org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_LANGUAGE
import org.keynote.godtools.android.db.Contract.FollowupTable.COLUMN_NAME

internal object FollowupMapper : BaseMapper<Followup>() {
    override fun mapField(values: ContentValues, field: String, obj: Followup) {
        when (field) {
            COLUMN_NAME -> values.put(field, obj.name)
            COLUMN_EMAIL -> values.put(field, obj.email)
            COLUMN_LANGUAGE -> values.put(field, serialize(obj.languageCode))
            COLUMN_DESTINATION -> values.put(field, obj.destination)
            COLUMN_CREATE_TIME -> values.put(field, serialize(obj.createTime))
            else -> super.mapField(values, field, obj)
        }
    }

    override fun newObject(c: Cursor) = Followup()
    override fun toObject(c: Cursor) = super.toObject(c).apply {
        name = c.getString(COLUMN_NAME)
        email = c.getString(COLUMN_EMAIL)
        languageCode = c.getLocale(COLUMN_LANGUAGE)
        destination = c.getLong(COLUMN_DESTINATION)
        createTime = getDate(c, COLUMN_CREATE_TIME)
    }
}
